### PR TITLE
NFT rewards on project page

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

This was originally intended to target #1356. My fault for being hasty with it, ended up doing a lot of work on this branch before we had finalised #1356. After I got feedback on that one, I figured it was more efficient just to implement the feedback on this branch rather than deal with massive merge conflicts. 

https://user-images.githubusercontent.com/96150256/178402075-74bdf20c-fe8f-423f-b00b-af57ae6b5c1c.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        className={className}
+        href={isLink ? nftReward.externalLink : undefined}
+      >
+        {nftReward.name}
+      </ExternalLink>
+      {nftReward.description ? (
+        <Tooltip title={nftReward.description}>{nftImage}</Tooltip>
+      ) : (
+        nftImage
+      )}
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx:64]: **suggestion (non-blocking)** there's a slightly nicer and more performant implementation here.
```suggestion
      <Tooltip title={nftReward.description} visible={nftReward.description ? undefined : false}>{nftImage}</Tooltip>
```

This says:
- when there's a description, set to `undefined`. The tooltip will behave as normal.
- When no description, set to `false`. The tooltip will never appear.

No big, just something to keep in mind for future.

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx [hunk 2]
```diff
@@ -160,7 +177,8 @@ export function V2ConfirmPayModal({
       onCancel={onCancel}
       confirmLoading={loading}
       width={640}
-      centered={true}
+      centered
+      destroyOnClose
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx:181]: **praise**

File: src/components/Loading.tsx [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+import { CSSProperties } from 'react'
 
-export default function Loading({ size }: { size?: number }) {
+export default function Loading({
+  size,
+  style,
+}: {
+  size?: number
+  style?: CSSProperties
+}) {
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/Loading.tsx:11]: Sure thang

File: src/components/inputs/Pay/PayInputGroup.tsx [hunk 4]
```diff
@@ -32,6 +34,8 @@ export default function PayInputGroup({
   weightingFn,
   disabled,
 }: {
+  payAmount: string
+  setPayAmount: (payAmount: string) => void
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/inputs/Pay/PayInputGroup.tsx:38]: Great call

File: src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-end',
+        marginRight: '25px',
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx:28]: Good call

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  nftReward: NFTRewardTier
+}): JSX.Element {
+  const [imageLoading, setImageLoading] = useState<boolean>(true)
+
+  const nftImage = (
+    <>
+      {imageLoading ? (
+        <LoadingOutlined size={25} style={{ marginLeft: 15 }} />
+      ) : null}
+      <div style={{ marginLeft: 15 }}>
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx:20]: Good call, done

File: src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        cursor: 'pointer',
+        transition: 'all 0s',
+      }}
+      onClick={onClick}
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx:32]: Got it thanks

File: src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+import { MouseEventHandler, useContext } from 'react'
+
+export function RewardTier({
+  rewardTier,
+  nextRewardTier,
+  isSelected,
+  onClick,
+}: {
+  rewardTier: NFTRewardTier
+  nextRewardTier: NFTRewardTier | undefined
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/RewardTier.tsx:14]: Sure, good call

File: src/components/v2/V2Project/NFTRewardsSection/index.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import { getNFTRewardTier, MOCK_NFTs } from 'utils/v2/nftRewards'
+
+import { RewardTier } from './RewardTier'
+
+export function NFTRewardsSection({
+  payAmount,
+  setPayAmount,
+}: {
+  payAmount: string
+  setPayAmount: (payAmount: string) => void
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/index.tsx:12]: Gone with `onPayAmountChange`

File: src/components/v2/V2Project/NFTRewardsSection/index.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+  const nftRewardTiers = MOCK_NFTs
+
+  if (!nftRewardTiers || nftRewardTiers.length < 1) return null
+
+  const renderRewardTier = (rewardTier: NFTRewardTier, index: number) => {
+    const isSelected = index === selectedIndex
+    return (
+      <RewardTier
+        key={index}
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/index.tsx:28]: Solid, done

File: src/components/v2/V2Project/NFTRewardsSection/index.tsx [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        rewardTier={rewardTier}
+        nextRewardTier={nftRewardTiers[index + 1]}
+        isSelected={isSelected}
+        onClick={e => {
+          e.stopPropagation()
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/index.tsx:33]: Nope, outgrowth. Removed

File: src/components/v2/V2Project/NFTRewardsSection/index.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+          setSelectedIndex(isSelected ? undefined : index)
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/index.tsx:34]: I think it's nice and worth leaving

File: src/components/v2/V2Project/NFTRewardsSection/index.tsx [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+            tip={t`Receive an NFT for contributing above a certain amount.`}
+          />
+        }
+      /> */}
+      <span style={{ fontSize: 12 }}>+NFT</span>
+      <div
+        style={{
+          display: 'flex',
+        }}
+      >
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/NFTRewardsSection/index.tsx:72]: Done

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx:64]: Great, thanks for this. I've used it

Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/NftRewardCell.tsx:64]: Used it for now anyway though

File: src/hooks/v2/nftRewards/IPFSToNFTRewards.ts [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+// Calls a cloudfunction to upload to IPFS created by @tankbottoms
+// Returns cid which points to where this NFT data is stored on IPFS
+export default function useNFTRewards(cid: string | undefined) {
+  return useQuery('nft-rewards', async () => {
+    if (!cid) {
+      return
+    }
+    const url = ipfsCidUrl(cid)
+    const response = await axios.get(url)
+    return transformNftRewardsData(response.data)
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/hooks/v2/nftRewards/IPFSToNFTRewards.ts:36]: Replaced all non-UI `NFT`s in code with `Nft`

**Comment by johnnyd-eth** [comment]
Thanks for the great review @aeolianeth. Resolved/commented on all

**Pull request merged**